src/pages/plant-groups/index.js

```diff
@@ -0,0 +1,148 @@
+import GridToolbar from 'src/components/Shared/GridToolbar'
+
+import Tree from 'src/components/Shared/Tree'
+import { RequestsContext } from 'src/providers/RequestsContext'
+import { SystemRepository } from 'src/repositories/SystemRepository'
+import { useWindow } from 'src/windows'
+
+// ** Windows
+import PlantWindow from './Windows/PlantGroupsWindow'
+
+// ** Helpers
+import ErrorWindow from 'src/components/Shared/ErrorWindow'
+import { useInvalidate, useResourceQuery } from 'src/hooks/resource'
+
+// ** Resources
+import { ResourceIds } from 'src/resources/ResourceIds'
+import { HighlightTwoTone } from '@mui/icons-material'
+
+const Plant = () => {
+  const { getRequest, postRequest } = useContext(RequestsContext)
+
+  const [selectedRecordId, setSelectedRecordId] = useState(null)
+
+  //states
+  const [windowOpen, setWindowOpen] = useState(false)
+
+  const [errorMessage, setErrorMessage] = useState(null)
+
+  const { stack } = useWindow()
+
+  async function fetchGridData(options = {}) {
+    const { _startAt = 0, _pageSize = 50 } = options
+
+    return await getRequest({
+      extension: SystemRepository.PlantGroup.qry,
+      parameters: `_startAt=${_startAt}&_pageSize=${_pageSize}&filter=`
+    })
+  }
+
+  const {
+    query: { data },
+    labels: _labels,
+    paginationParameters,
+    refetch,
+    access
+  } = useResourceQuery({
+    queryFn: fetchGridData,
+    endpointId: SystemRepository.PlantGroup.qry,
+    datasetId: ResourceIds.PlantGroups
+  })
+
+  const invalidate = useInvalidate({
+    endpointId: SystemRepository.PlantGroup.qry
+  })
+
+  const columns = [
+    {
+      field: 'reference',
+      headerName: _labels.reference,
+      flex: 1
+    },
+    {
+      field: 'name',
+      headerName: _labels.name,
+      flex: 1
+    },
+
+    {
+      field: 'parentName',
+      headerName: _labels.parent,
+      flex: 1
+    }
+  ]
+
+  const add = () => {
+    setWindowOpen(true)
+  }
+
+  const edit = obj => {
+    setSelectedRecordId(obj.recordId)
+    setWindowOpen(true)
+  }
+
+  const del = async obj => {
+    await postRequest({
+      extension: SystemRepository.PlantGroup.del,
+      record: JSON.stringify(obj)
+    })
+    invalidate()
+    toast.success('Record Deleted Successfully')
+  }
+
+  function onTreeClick() {
+    stack({
+      Component: Tree,
+
+      props: {
+        data: data
+      },
+      width: 500,
+      height: 400,
+      title: 'Tree'
+    })
+  }
+
+  return (
+    <>
+      <Box>
+        <GridToolbar onAdd={add} maxAccess={access} onTree={onTreeClick} />
+        <Table
+          columns={columns}
+          gridData={data}
+          rowId={['recordId']}
+          onEdit={edit}
+          onDelete={del}
+          isLoading={false}
+          pageSize={50}
+          refetch={refetch}
+          paginationParameters={paginationParameters}
+          paginationType='api'
+          maxAccess={access}
+        />
+      </Box>
+      {windowOpen && (
+        <PlantWindow
```

Comment:
No need for PlantComponentWindow delete it and open PlantGroupsForm directly by using useWindow

Reply:
done